common/extensions/api/bat_notifications.json

```diff
@@ -0,0 +1,119 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+[
+  {
+    "namespace": "batNotifications",
+    "description": "Use the <code>chrome.batNotifications</code> API to receive BAT notifications.",
+    "compiler_options": {
+      "implemented_in": "brave/browser/extensions/api/bat_notifications_api.h"
+    },
+    "events": [
+      {
+        "name": "onNotificationAdded",
+        "type": "function",
+        "description": "Fired when a notification is added",
+        "parameters": [
+           {
+             "name": "id",
+             "type": "integer"
+           },
+           {
+             "name": "type",
+             "type": "integer"
+           },
+           {
+             "name": "timestamp",
+             "type": "integer"
+           }
+	 ]
+      },
+      {
+        "name": "onNotificationDeleted",
+        "type": "function",
+        "description": "Fired when a notification is deleted",
+        "parameters": [
+           {
+             "name": "id",
+             "type": "integer"
+           },
+           {
+             "name": "type",
+             "type": "integer"
+           },
+           {
+             "name": "timestamp",
+             "type": "integer"
+           }
+	 ]
+      },
+      {
+        "name": "onAllNotificationsDeleted",
+        "type": "function",
+        "description": "Fired when all notifications are deleted",
+        "parameters": []
+      },
+      {
+        "name": "onGetNotification",
+        "type": "function",
+        "description": "Fired when a notification is retrieved",
+        "parameters": [
+           {
+             "name": "id",
+             "type": "integer"
+           },
+           {
+             "name": "type",
+             "type": "integer"
+           },
+           {
+             "name": "timestamp",
+             "type": "integer"
+           }
+	 ]
+      }
+    ],
+    "functions": [
```

Comment:
Wasn't sure if all of these functions would be necessary? But added them in case, let me know.